lib/sailthru/client.rb

```diff
@@ -317,6 +317,27 @@ def receive_optout_post(params, request)
     #   params, Hash
     #   request, String
     # returns:
+    #   TrueClass or FalseClass, Returns true if the incoming request is an authenticated list post.
+    def receive_list_post(params, request)
+      if request.post?
+        [:action, :email, :sig].each { |key| return false unless params.has_key?(key) }
+
+        return false unless params[:action] == 'update'
+
+        sig = params.delete(:sig)
+        params.delete(:controller)
+        return false unless sig == get_signature_hash(params, @secret)
+        return true
```

Comment:
This could probably be refactored to `sig != get_signature_hash(params, @secret)` instead of performing an if statement and then forcing a `return`.